proxy-backend: Add info to the error message which proxy target isn't well configured

jerolimov opened the pull request:
I got an error message when I tried a backstage instance without some required configs. The `proxy-backend` fails with the error message bwloe and I didn't know which of the different proxies was the reason.

This PR only makes the error message a bit more verbose:

Error message change:

```diff
-backend:start: Backend failed to start up Error: Proxy target must be a string
+backend:start: Backend failed to start up Error: Proxy target for route "/my-plugin-route" must be a string, but is undefined
backend:start:     at buildMiddleware (..../node_modules/@backstage/plugin-proxy-backend/dist/index.cjs.js:35:11)
```

freben commented:
Hi! This needs a changeset to be released (see the comment above and the build details below)

jerolimov commented:
> Hi! This needs a changeset to be released (see the comment above and the build details below)

Thanks @freben, since this was my first PR: I saw the message from the bot, but wasn't sure if it's rly required for an error message.

But I have applied the change now, and also fixed the prettier issue and updated a unit test as well. Let's see if the tests pass now. :crossed_fingers:
